From 38cdc7afc82df3c246daa7e3b12105dcf1376fdf Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 17 Aug 2018 10:59:17 +0800
Subject: [PATCH 4525/5242] MLK-19413-14 gpu: imx: dpu: framegen: Add
 framegen_is_master/slave() helper support

commit  d8a0a5bcc7d907dfe1a25c5149d7260f55907351 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds framegen_is_master/slave() helpers support so that
callers may know if a framegen is a master or slave framegen.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-framegen.c |   12 ++++++++++++
 include/video/dpu.h                |    2 ++
 2 files changed, 14 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-framegen.c b/drivers/gpu/imx/dpu/dpu-framegen.c
index 581b155..d15189a 100644
--- a/drivers/gpu/imx/dpu/dpu-framegen.c
+++ b/drivers/gpu/imx/dpu/dpu-framegen.c
@@ -552,6 +552,18 @@ void framegen_disable_clock(struct dpu_framegen *fg)
 }
 EXPORT_SYMBOL_GPL(framegen_disable_clock);
 
+bool framegen_is_master(struct dpu_framegen *fg)
+{
+	return fg->id == 0;
+}
+EXPORT_SYMBOL_GPL(framegen_is_master);
+
+bool framegen_is_slave(struct dpu_framegen *fg)
+{
+	return fg->id == 1;
+}
+EXPORT_SYMBOL_GPL(framegen_is_slave);
+
 struct dpu_framegen *dpu_fg_get(struct dpu_soc *dpu, int id)
 {
 	struct dpu_framegen *fg;
diff --git a/include/video/dpu.h b/include/video/dpu.h
index edd66f3..c9bd78f 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -619,6 +619,8 @@ void framegen_read_timestamp(struct dpu_framegen *fg,
 void framegen_wait_for_frame_counter_moving(struct dpu_framegen *fg);
 void framegen_enable_clock(struct dpu_framegen *fg);
 void framegen_disable_clock(struct dpu_framegen *fg);
+bool framegen_is_master(struct dpu_framegen *fg);
+bool framegen_is_slave(struct dpu_framegen *fg);
 struct dpu_framegen *dpu_fg_get(struct dpu_soc *dpu, int id);
 void dpu_fg_put(struct dpu_framegen *fg);
 struct dpu_framegen *dpu_aux_fg_peek(struct dpu_framegen *fg);
-- 
1.7.9.5

